src/os/bluestore/BlueStore.cc

```diff
@@ -6515,7 +6486,22 @@ int BlueStore::_upgrade_super()
     // - super: added ondisk_format
     // - super: added min_readable_ondisk_format
     // - super: added min_compat_ondisk_format
+    // - super: added min_alloc_size
+    // - super: removed min_min_alloc_size
     KeyValueDB::Transaction t = db->get_transaction();
+    {
+      bufferlist bl;
+      db->get(PREFIX_SUPER, "min_min_alloc_size", &bl);
+      auto p = bl.begin();
+      try {
+	::decode(min_alloc_size, p);
```

Comment:
Suggest to decode using a standalone variable instead of min_alloc_size. With an explicit type specification for that var (equal to min_min_alloc_size). Otherwise one might have a well hidden issue when changing min_alloc_size type in future.